### PR TITLE
Ghost bootloader

### PR DESCRIPTION
**Pull request by brandonrc**
Working on a bootloader for ghost tx module.

### CODE REVIEW

**Comment by brandonrc** [comment]
@peho-nordic Need some help!

**Comment by brandonrc** [comment]
@cruwaller need some help with the bootloader. 

Thanks, 

Currently I am not getting into the ELRS firmware at boot. Just gets stuck.

File: src/bootloader/src/Src/uart.c [hunk 1]
```diff
@@ -407,6 +407,7 @@ static uint8_t uart_valid_pin_tx(uint32_t pin)
   switch (pin) {
     case GPIO('A', 2):
     case GPIO('A', 9):
+    case GPIO('A', 10):
```
Comment (cruwaller) [inline src/bootloader/src/Src/uart.c:410]: STM32F3xx only.

File: src/bootloader/src/Src/uart.c [hunk 2]
```diff
@@ -528,6 +529,12 @@ void uart_init(uint32_t baud, uint32_t pin_rx, uint32_t pin_tx, int32_t duplexpi
     pin_rx = 0;
 #endif
 
+  // if (halfduplex) {
+    #if defined(STM32F3xx) && defined(UART_INV)
+      USART1->CR2 |= USART_CR2_RXINV | USART_CR2_TXINV | USART_CR2_SWAP; //inv
```
Comment (cruwaller) [inline src/bootloader/src/Src/uart.c:534]: Write there bits into `UART_CR_RX` and `UART_CR_TX` variables.

File: src/bootloader/src/platformio.ini [hunk 3]
```diff
@@ -368,3 +368,21 @@ build_flags =
...
+
+[env:GHOST_TX]
+board = stm32f303
+board_upload.maximum_size = 16384
+build_flags =
+    -D UART_INV
+    -D MCU_TYPE=GHOST_TX
+    -D WS2812_LED_PIN="B,6"
+    -D UART_TX_PIN="A,10"
+    -D UART_RX_PIN="A,10"
```
Comment (cruwaller) [inline src/bootloader/src/platformio.ini:380]: Remove RX definition.

File: src/bootloader/src/Src/uart.c (hunk 2 above)
Comment (cruwaller) [reply src/bootloader/src/Src/uart.c:534]: And `USART_CR2_SWAP` if TX pins is physical uart rx pin (e.g PA10).

**Review by cruwaller — CHANGES_REQUESTED** [review]

**Review by peho-nordic — APPROVED** [review]
This bootloader is tested to work. Thanks @brandonrc !

**Review by cruwaller — APPROVED** [review]

**Pull request merged**